network/src/primary.rs

```diff
@@ -263,3 +107,117 @@ impl UnreliableNetwork for PrimaryToWorkerNetwork {
         handler
     }
 }
+
+pub struct PrimaryNetwork {
+    network: anemo::Network,
+    retry_config: RetryConfig,
+    /// Small RNG just used to shuffle nodes and randomize connections (not crypto related).
+    rng: SmallRng,
+    // One bounded executor per address
+    executors: HashMap<anemo::PeerId, BoundedExecutor>,
+}
+
+impl PrimaryNetwork {
+    pub fn new(network: anemo::Network) -> Self {
```

Comment:
What are your thoughts on handling the `anemo::Network` construction in this method and only passing the `NetworkIdentity` as arguments? i.e.

`pub fn new(myself: NetworkIdentity, peers: Vec<NetworkIdentity>) -> Self {`

```
struct NetworkIdentity {
    key_pair: KeyPair,
    address: MultiAddr
}
```

Reply:
I don't think that'll work since we only want a single anemo network instance where are multiple components all have their own "PrimaryNetwork" handles. Some point soon we'll also have the other interfaces all operating on top of a single instance of an anemo network so that will need to be built prior to this so that it can be shared.